### DISCUSSION TITLE
Text Height

### DISCUSSION THREAD

**Discussion by Nazar1ky** [pymupdf/PyMuPDF#4149] [Q&A] [answered]
Hello, I read docs but I get confused how to get text height. Here is my representation code:

```python
import pymupdf

with pymupdf.open() as doc:
    page = doc.new_page(height=200, width=200)

    # Load Segoe UI fonts
    font_bold = pymupdf.Font(fontfile="C:/Windows/Fonts/segoeuib.ttf")
    page.insert_font(fontname="segoeuib", fontbuffer=font_bold.buffer)

    fs = 10
    height = (font_bold.ascender - font_bold.descender) * fs
    page.insert_text((0, height), "ÊQXxbdpqgy(@_`[]{})⭐🤣", fontname="segoeuib", fontsize=fs)

    doc.save("test.pdf")
```

The result I got is not ideally at 0, 0. Maybe I doing something wrong?
![image](https://github.com/user-attachments/assets/b81f1535-225c-4c12-9fd7-8de6e73f4e17)
Thanks!

**Comment by JorjMcKie** [comment, marked as answer]
The `(0, height)` is the insertion point = "origin"! This is higher up than the bbox.y1.
Use `(0, font.ascender * fs)` instead.

**Reply by Nazar1ky** [reply]
Thank you so much for answer, it works!

**Reply by Nazar1ky** [reply]
But for Emoji font it works a bit different, is there way to get bbox?

**Reply by JorjMcKie** [reply]
Why should it be different?

**Reply by Nazar1ky** [reply]
Hi. Sorry for late answer. 
![image](https://github.com/user-attachments/assets/1f7ef12d-36aa-415a-a113-49b86a4287d5)

```Python
font_emoji = pymupdf.Font(fontfile="C:/Windows/Fonts/seguiemj.ttf")
page.insert_font(fontname="segoeuimj", fontbuffer=font_emoji.buffer)

page.insert_text((0, font_emoji.ascender * fs), "ÊQXxbdpqgy(@_`[]{})⭐🤣", fontname="segoeuimj", fontsize=fs, color=(0.5,))
```
The Ê is not showed fully, and emoji too.

**Reply by JorjMcKie** [reply]
That's the fault of the font maker.

**Reply by Nazar1ky** [reply]
Alright thank you!
